shared/src/TwinCollectionArray.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Devices.Shared
+{
+    /// <summary>
+    /// Represents a property array in <see cref="TwinCollection"/>
+    /// </summary>
+    public class TwinCollectionArray : JArray
+    {
+        private const string MetadataName = "$metadata";
+        private const string LastUpdatedName = "$lastUpdated";
+        private const string LastUpdatedVersionName = "$lastUpdatedVersion";
+
+        private readonly JObject _metadata;
+
+        internal TwinCollectionArray(JArray jArray, JObject metadata)
+            : base(jArray)
+        {
+            _metadata = metadata;
+        }
+
+        /// <summary>
+        /// Gets the value for the given property name
+        /// </summary>
+        /// <param name="propertyName">Property Name to lookup</param>
+        /// <returns>Value if present</returns>
+        public dynamic this[string propertyName]
+        {
+            get
+            {
+                if (propertyName == MetadataName)
+                {
+                    return GetMetadata();
+                }
+
+                if (propertyName == LastUpdatedName)
+                {
+                    return GetLastUpdated();
+                }
+
+                if (propertyName == LastUpdatedVersionName)
+                {
+                    return GetLastUpdatedVersion();
+                }
+
+                throw new ArgumentException($"'Newtonsoft.Json.Linq.JArray' does not contain a definition for '{propertyName}'.");
```

Comment:
is it really the Newtonsoft object that doesn't have a definition for the property? or the twin object?

Reply:
yeah, it should be ~TwinCollectionArray` actually, i'll update it